zapcore/sampler.go

```diff
@@ -81,17 +81,87 @@ func (c *counter) IncCheckReset(t time.Time, tick time.Duration) uint64 {
 	return 1
 }
 
+// SamplingDecision represents a decision made by sampler.
+type SamplingDecision uint8
```

Comment:
I'm guessing this is so that it takes one byte instead of 8?

Reply:
yup, as byte is an alias to uint8.